src/hapi.erl

```diff
@@ -170,7 +171,10 @@ req(Method, URI0, Opts) ->
                    Req0#{method => HTTPMethod, body => Body};
                _ -> Req0#{method => Method}
            end,
-    req(Req1, Host, Families, Port, DeadLine, ReqTimeout, {RetryTimeout, 0, MaxRetries}).
+    trace_request(Req1, Opts),
```

Comment:
Lets add a request_id - a number to match response and request. 

While we do log PIDs - they are often reused in BEAM, so if we are looking for a particular response we might get multiple log records which we have to analyze. It is not a challenge, but easily avoidable extra work.

It doesnt have to be anything fancy, just uniform random number would be ok, like `unique_integer/0`  or `os:system_time`

Reply:
Discussed in voice. Comment discarded.

Reply:
Done